src/unitxt/api.py

```diff
@@ -138,11 +139,18 @@ def infer(
     engine,
     dataset_query: Optional[str] = None,
     return_data=False,
+    infer_log_probs=False,
     **kwargs,
 ):
     dataset = produce(instance_or_instances, dataset_query, **kwargs)
     engine, _ = fetch_artifact(engine)
-    raw_predictions = engine.infer(dataset)
+    if infer_log_probs:
+        raw_predictions = engine.infer_log_probs(dataset)
```

Comment:
no one guarantee you the engine implemented the interface. You need to check it

Reply:
done